### DISCUSSION TITLE
NEXT_PUBLIC and Docker

### DISCUSSION THREAD

**Discussion by jkupcho** [vercel/next.js#28398] [Help]
Hello!

I'm trying to use `NEXT_PUBLIC_*` to drive client side environment variables for things like OAuth Client IDs. The problem I'm running into is these IDs change between environments the application is deployed to, which it seems like by default would mean building multiple Docker images.

I'd like to take advantage of the static code generation, if possible, and avoid the Runtime configuration when building the Docker image as that's going to force server side renders for many of the pages.

Anyone have any tips or tricks, maybe building multiple static sites then pushing to S3 somehow, or am I stuck with either multiple Docker images, or using the Runtime configuration.

Thanks!

**Comment by rafalbromirski** [comment]
Welcome to the rabbit hole.

You could probably use:
- https://github.com/andrewmclagan/react-env

Also make sure you check discussion for "docker env" because there is a plenty of people who struggle with it as well

**Comment by rafalbromirski** [comment]
https://github.com/vercel/next.js/discussions/28759

This might be useful as well
